tests/5.1/requires/test_target_is_accessible_with_usm.c

```diff
@@ -0,0 +1,68 @@
+//--------------- test_target_is_accessible_with_usm.c---------------------//
+//
+// OpenMP API Version 5.1 Aug 2021
+//
+// This test checks the omp_target_is_accessible device routine.
+// In this test the output of the target_is_accessible call should return
+// true because the requires unified shared memory is used.
+//-----------------------------------------------------------------------//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+#include <math.h>
+
+#define N 100
+
+#pragma omp requires unified_shared_memory
+
+int check_device(){
+  int errors = 0;
+  int check_test = -57;
+  const int buf_size = sizeof(int) * N;
+  const int dev = omp_get_default_device();
+
+  int *ptr = (int *) malloc(buf_size);
+
+  if(ptr == NULL){
+    OMPVV_INFOMSG("Memory allocation on host failed");
+    return 1;
+  }
+  for(int i = 0;i < N; i++)
+    ptr[i] = i;
+
+  check_test = omp_target_is_accessible(ptr, buf_size, dev);
+
+  OMPVV_TEST_AND_SET_VERBOSE(errors, check_test == 0);
+
+  OMPVV_ERROR_IF(check_test == -57, "Call to omp_target_is_accessible did not return true or false");
+  OMPVV_INFOMSG_IF((check_test != 0 && check_test != -57), "The host memory is accessible from the default device.");
```

Comment:
The first line lines actually check whether the assignment `check_test = omp_target_is_accessible(...)` on the host works – and _true_ is not `-57`.

I think there is no reason to assume that the call to the API function returned to the caller but that the assignment did not work – and as the assignment is not done done by reference but as return value, even a `return;` instead of `return res;` in the function will not keep the original value. — Additionally, -57 matches the OpenMP definition of _true_.

Likewise for the second line. Thus, I would do:

```suggestion
  OMPVV_INFOMSG_IF(check_test != 0, "The host memory is accessible from the default device.");
```